src/main/java/com/smoothstack/utopia/counter/service/CustomerService.java

```diff
@@ -33,9 +33,7 @@ public void saveCustomer(Customer customer) {
 	
 	public Customer readCustomer(Integer customerId) throws InvalidIdException {
 		Optional<Customer> customer = customerRepo.findById(customerId);
-		if (!customer.isPresent()) {
-			throw new InvalidIdException("That ID is invalid.");
-		}
+		customer.orElseThrow(() -> new InvalidIdException("That ID is invalid."));
```

Comment:
i think you can chain orElseThrow directly to return object without using get and will return object inside optional